docs/eeps/draft-obsolete-execution-as-a-command.md

```diff
@@ -0,0 +1,57 @@
+---
+EEP: unnumbered
+Title: Obsolete Execution as a Command
+Author: dmikurube
+Status: Draft
+Type: Standards Track
+---
+
+Obsolete Execution as a Command
+================================
+
+Background
+-----------
+
+Embulk has provided a method to run it as a simple command `embulk` on the command line as illustrated by the examples below.
+
+```
+$ embulk run postgresql-example.yml
+...
+```
+
+```
+$ embulk gem install embulk-input-s3
+...
+```
+
+```
+$ embulk mkbundle bundle_dir
+...
+```
+
+It was implemented by embedding a shell script since [v0.4.6](https://github.com/embulk/embulk/commit/38b000d5c6be13ad4ea2818d3cf6ea67e9d86c51) (a batch file for Windows since [v0.6.11](https://github.com/embulk/embulk/pull/192)) in Embulk's distributed "executable" JAR files to emit a `java` command with some command-line options configured.
+
+It worked well as long as Embulk expected almost the singular version of Java, which was Oracle's Java 7 (up to Embulk v0.8), or [Java 8 (since Embulk v0.9)](https://github.com/embulk/embulk/pull/853).
+
+However, Java has changed their versioning and release strategy since Java 9. Java has much more frequent major releases twice a year, and a variety of Java runtimes are released by many organizations, such as [Oracle's](https://www.oracle.com/java/), [Red Hat's](https://developers.redhat.com/products/openjdk/overview), [Azul's Zulu](https://www.azul.com/products/core/), [Adoptium's Eclipse Temurin](https://adoptium.net/temurin/), [Amazon's Corretto](https://aws.amazon.com/en/corretto/), Ubuntu's own builds, and more.
+
+Unfortunately, they differ from each other. Different versions have different command-line options obviously. Furthermore, even within the same version, different distributions may have different command-line options and behaviors.
+
+As discussed in [EEP-8: Milestones to Embulk v1.0, and versioning strategies to follow](./eep-0008.md) and [EEP-11: Moving Forward Together with Modern Java](./eep-0011.md), Embulk would target a range of Java versions, not just a single version at a time. Also, needless to say, Embulk should not limit the Java distributions it runs on.
+
+The embedded shell script (and batch file) would still need to cover those variations of Java versions and distributions, but there are too many Java variations in the world to cover, and shell scripts (and batch files) are not so powerful to cover such complexity. On the other hand, depending on another scripting runtime (such as Ruby or Python) would add another complex dependency to the Embulk ecosystem.
+
+Embulk's use-cases in the wild
+-------------------------------
+
```

Comment:
Is this doc still a work in progress? It seems some sentences will be added in this part later.